syntax = "proto3";

package custom_node;

service CustomNodeHandler {
    rpc StartSession(StartSessionRequest) returns (StartSessionResponse) {}
    rpc ProcessFrame (ProcessFrameRequest) returns (ProcessFrameResponse) {}
}

message Frame {
    float  timestamp   = 3;
    string frame_id    = 4;
    string instance_id = 5;
    string skill_id    = 6;
    string device_id   = 7;
    string datetime    = 8;
}


message StartSessionRequest {
    int64 seq = 1;
}

message StartSessionResponse {
    int64 ack = 1;
}

message ProcessFrameRequest {
    int64 seq = 1;
    Frame frame = 2;
}

message ProcessFrameResponse {
    int64 ack = 1;
    Frame frame = 2;
}

